packages/react-jss/src/createUseStyles.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+
+import React from 'react'
+import {ThemeContext as DefaultThemeContext} from 'theming'
+
+import JssContext from './JssContext'
+import {
+  createStaticSheet,
+  addDynamicRules,
+  removeDynamicRules,
+  updateDynamicRules
+} from './utils/sheets'
+import {getIndex} from './utils/indexCounter'
+import type {HookOptions, Styles} from './types'
+import {unmanageSheet, manageSheet} from './utils/managers'
+import {getSheetClasses} from './utils/getSheetClasses'
+
+const noTheme = {}
+
+const createUseStyles = <Theme: {}>(styles: Styles<Theme>, options?: HookOptions<Theme> = {}) => {
+  const {index = getIndex(), theming, name = 'Hook', ...sheetOptions} = options
+  const ThemeContext = (theming && theming.context) || DefaultThemeContext
+  const useTheme =
+    typeof styles === 'function'
+      ? // $FlowFixMe
+        (): Theme => React.useContext(ThemeContext)
+      : // $FlowFixMe
+        (): Theme => noTheme
+
+  return (data: any) => {
+    const context = React.useContext(JssContext)
+    const theme = useTheme()
+
+    // When the theme or the context changes we create a new sheet
+    const sheet = React.useMemo(
```

Comment:
Do I understand correctly that we rely on memo to reuse the static sheet between component instances?